orca-clouddriver/src/main/groovy/com/netflix/spinnaker/orca/clouddriver/tasks/providers/kubernetes/KubernetesContainerFinder.groovy

```diff
@@ -63,6 +63,15 @@ class KubernetesContainerFinder {
 
     containers.forEach { container ->
       if (container.imageDescription.fromContext) {
+	//  If there are parallel Find images stages , then deploymentDetails would be fetched from it. 
```

Comment:
@ajordens : 

Pipeline i used:  two parallel find images with  deploy as intermediate stage.
![image](https://user-images.githubusercontent.com/24909748/26916272-4d3655fa-4c47-11e7-9428-ade5cf7b8db2.png)

here pattern is     .* 

Find image save the stage context using  ExecutionContextManager
and it is having 
 static class DelegatingHashMap implements Map<String, Object> 


and find image write to local , as well as global context. 
using these lines 

// fetch the current stage (w/ global context merged in)
        stage = currentStage(chunkContext, true)

I would investigate  further according to the above comments  to look the area where global context is overridden. 
Regards,
Jatin bajaj